sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1310,6 +1303,14 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
         "quantile",
     )
 
+    _parameter_constraints = {**BaseHistGradientBoosting._parameter_constraints}
+    _parameter_constraints.update(
+        {
+            "loss": [StrOptions(set(_VALID_LOSSES)), BaseLoss],
```

Comment:
We can mark the deprecated options like this:
```suggestion
            "loss": [StrOptions(set(_VALID_LOSSES), deprecated={"least_squares", "least_absolute_deviation"}), BaseLoss],
```

Reply:
Note that accepting this suggestion via GitHub will require to add a new commit to reformat with black.